Revising contribution docs (Lombiq Technologies: OCORE-154) (#15706)
OrchardCMS/OrchardCore: src/docs/guides/contributing/contributing-code.md

3. Ensure `OrchardCore.Cms.Web` is set as the startup project. Then run the app with Ctrl+F5.

## Selecting what to work on

We warmly welcome fixes and features! Here are some resources to help you get started on how to contribute code:

- Check out [the issue tracker](https://github.com/OrchardCMS/OrchardCore/issues) for open bug reports and feature requests. Prefer to select issues [scheduled for the upcoming release](https://github.com/OrchardCMS/OrchardCore/milestones). You can [sort issues by most commented](https://github.com/OrchardCMS/OrchardCore/issues?q=is%3Aissue+is%3Aopen+sort%3Acomments-desc) and [most thumbs up](https://github.com/OrchardCMS/OrchardCore/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) (as well as other reactions similarly). These correlate with popularity, i.e. we can see what the community most wants. 
- ["Good first issue" issues](https://github.com/OrchardCMS/OrchardCore/labels/good%20first%20issue): We think these are a good for newcomers.
- ["Help wanted" issues](https://github.com/OrchardCMS/OrchardCore/labels/help%20wanted): These issues are up for grabs. Comment on an issue if you want to create a fix.

## Identifying the scale

> First identify the scale of what you would like to contribute. If it is small (grammar/spelling or a bug fix) feel free to start working on a fix. If you are submitting a feature or other substantial code contribution, please discuss it with the team first and ensure it follows the roadmap and fits into the bigger picture. We'd hate to see your work go to waste! The best is if you work on open issues, and open an issue for your idea too, before starting coding.

Reviewer: ```suggestion
First, identify the scale of what you would like to contribute. If it is small (grammar/spelling or a bug fix) feel free to start working on a fix. If you are submitting a feature or other substantial code contribution, please discuss it with the team first and ensure it follows the roadmap and fits into the bigger picture. We'd hate to see your work go to waste! The best is if you work on open issues, and open an issue for your idea too, before starting coding.
```

> You might also read these two blogs posts on contributing code: [Open Source Contribution Etiquette](http://tirania.org/blog/archive/2010/Dec-31.html) by Miguel de Icaza and [Don't "Push" Your Pull Requests](https://www.igvita.com/2011/12/19/dont-push-your-pull-requests/) by Ilya Grigorik.

Reviewer: ```suggestion
You might also read these two blog posts on contributing code: [Open Source Contribution Etiquette](http://tirania.org/blog/archive/2010/Dec-31.html) by Miguel de Icaza and [Don't "Push" Your Pull Requests](https://www.igvita.com/2011/12/19/dont-push-your-pull-requests/) by Ilya Grigorik.
```

## Submitting a pull request (PR)

!!! info
    If you don't know what a pull request is, read [this article](https://help.github.com/articles/using-pull-requests).

- Familiarize yourself with the project and our coding conventions, as you can see in the repository. We follow the [ASP.NET Core Engineering guidelines](https://github.com/dotnet/aspnetcore/wiki/Engineering-guidelines).

> - Make sure the repository can build and all tests pass. This is also checked by the CI workflows.

Reviewer: ```suggestion
- Make sure the repository can be built and all tests pass. This is also checked by the CI workflows.
```

    When in doubt, just ask the community for guidance, we're happy to help!

## Feedback

All code submissions will be reviewed and tested by the core team, and only those that meet our bar for both quality and design/roadmap appropriateness will be merged. Please be patient; we very much appreciate you contributing to Orchard Core!

!!! note
    Following the below recommendations will ensure that your PR is merged as quickly as possible. Please work with us so it's a pleasant experience for everyone.

- Please update your pull request according to feedback until it is approved by one of the core team members.
- Apply [suggested changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes) directly so the reviewer doesn't have to eyeball the changes. These resolve themselves after applying them, and that's fine.
- Don't resolve other conversations so it's easier to track for the reviewer. Then, the reviewer will resolve them.
- Feel free to mark conversations that you addressed to keep track of them with an emoji or otherwise, just don't resolve them.

> - Please keep conversations happening in line comments in those convos, otherwise, communication will be a mess. If you have trouble finding them, see [this video](https://github.com/OrchardCMS/OrchardCore/pull/14749#issuecomment-1917976028).

Reviewer: ```suggestion
- Please keep conversations happening inline comments in those convos, otherwise, communication will be a mess. If you have trouble finding them, see [this video](https://github.com/OrchardCMS/OrchardCore/pull/14749#issuecomment-1917976028).
```

Author: I mean "line comments" here, i.e. comments added on lines under the PR, not "inline comments".
